packages/amplify_authenticator/lib/src/services/amplify_auth_service.dart

```diff
@@ -270,8 +270,15 @@ class AmplifyAuthService implements AuthService {
   }
 
   @override
-  Stream<AuthHubEvent> get hubEvents =>
-      Amplify.Hub.availableStreams[HubChannel.Auth]!.cast();
+  Stream<AuthHubEvent> get hubEvents async* {
+    // Auth channel will be null until configuration completes
+    await Amplify.asyncConfig;
+    await for (final event in Amplify.Hub.availableStreams[HubChannel.Auth]!) {
+      if (event is AuthHubEvent) {
```

Comment:
Was this emitting non-AuthHubEvent values?

Reply:
No. The type is `HubEvent` though, so it either has to be cast or filtered. In `next` we cast it. I could do that here as well. Filtering it just seemed safer.